app/controllers/folder_share_controller.rb

```diff
@@ -7,6 +7,10 @@ def show
     @folder = fetch_folder_by_token
     authorize share_data, policy_class: SharePolicy
 
+    search_results ||= FolderSignService.new(search:, relation: policy_scope(Sign), folder: @folder).process
```

Comment:
nit: there's no point conditionally assigning here

```suggestion
    search_results = FolderSignService.new(search:, relation: policy_scope(Sign), folder: @folder).process
```